# -*- coding: utf-8 -*-
# -*- mode: org -*-

#+TITLE:
#+LANGUAGE:  en
#+STARTUP: overview indent inlineimages logdrawer
#+OPTIONS: H:5 author:nil email:nil creator:nil timestamp:nil skip:nil toc:nil ^:nil
#+TAGS: noexport(n) deprecated(d)
#+EXPORT_SELECT_TAGS: export
#+EXPORT_EXCLUDE_TAGS: noexport

# #+LATEX_CLASS: IEEEtran
#+LaTeX_CLASS: article
# #+LaTeX_CLASS: acm-proc-article-sp

#+BABEL: :session *R* :cache yes :results output graphics :exports both :tangle yes 

* Introduction
* Materials and methods
** COMMENT Model
Following the common notations in linear latent factor regression models cite
here , for an observation $i \in \{1, ..., n\}$, we assume that the random
vector $G_i = (G_{i 1},..., G_{i L})$ assuming the co-variable $X_i = (X_{i 1},
..., X_q{i d})$ is a multivariate normal distribution such that
   
$$ E[G_i | X_i] = X_i B^T $$
   
where $B$ is the unknown regression coefficient. Then assuming $K$ latent
factors we write the covariance matrix $var(G_i|X_i) = \Sigma$ as follow
 
$$ \Sigma = D + V V^T $$ 
   
where $V$ is $L \times K$ matrix of latent
factor loadings and $D$ is the diagonal matrix of size $L$. We can write the
following matrix notation of the model: 
   
$$ G = U V^T + X B^T + E $$ 
   
where $U$ is a $n \times K$ matrix of latent factor scores and $E$ is the error
matrix distributed with a multivariate normal distribution with the diagonal
covariance matrix $D$.
** Model 
We consider a G matrix which record n observations of a vector $G_i = (G_{i
1},..., G_{i L})$ of size L. For each observation 
** Estimation
:LOGBOOK:
- Note taken on [2017-05-12 Ven 16:35] \\
  On va plutot faire uen partie pour l'estimateur lasso et un autre pour la ridge
  et une derniere pour les missing values.
:END:
To estimate model parameter we propose the following optimization problem 

$$min_{rk(C) \leq K} \frac{1}{2} ||G - C - X B^T||_{F}^2 + \lambda r(B)$$ 

where
$C$ is the latent matrix such that $C = U V^T$, $||.||_F$ the Frobenius matrix
,$rk(.)$ the function which returns the rank of a matrix and $r$ a convex
regularization function. In this article we discuss the ridge regularization
function $||.||^2_2$ and the lasso regularization function $||.||$.
*** A ridge regularized estimator
**** Analytic solution
*** A lasso regularized estimator
We write the following convex problem: 

$$min \frac{1}{2} ||G - C - X B^T||_{F}^2 + \lambda ||B|| + \gamma ||C||*)$$ 

where ... It is easy to prove that the problem is convex.

** Algorithm
   
In this section we propose algorithms to estimate latent matrices $U$ and $V$
and association parameter $B$. We present a very efficient algorithm for the
ridge regularized loss function based on analytic solutions of the
optimization problem. We also present an efficient alternated algorithm for
data with missing values and a general regularization convex function.


*** Missing values
It is frequent that there are missing values in the data matrix. A solution
is to use an imputation algoritm before running the association study. But
this can lead to bias estimation and spurious association pattern introduced
by the imputation algorithm. We propose an algorithm we avoid using a
preliminary imputation method.

** Hypothesis testing

** Simulations and data

*** Generative model simulation
We used equation to generate generative model dataset. The latent factor
scores and loadings $U$ and $V$ were generated using a multivariate gaussian
distribution with a zero mean and a $K$ identity matrix for the covariance
matrix where is the number of latent factor. The error matrix $E$ was
generated using a multivariate gaussian distribution with a zero mean and a
$L$ identity matrix for the covariance matrix where $L$ is the number of
variables. The co-variable $X$ was generated with a normal distribution with
the mean equal to zero and the standard deviation equal to one such that the
Pearson linear correlation between $X$ and $U_1$ the first latent score
matrix equal to $c$.

*** Real data example
In to evaluate our methods on real data we chose realized a genome-wide
association study (GWAS), an genome-wide association study (EWAS) and an ecological
association study (EAS). 
**** Association study of DNA methylation with rheumatoid arthritis (EWAS)
In order to evaluate the ability to our method to correct for unobserved
confounding variable we used data from a recent association study of DNA
methylation with rheumatoid arthritis (RA) cite:Liu_2013. For this data set
confounding variables (batch effect, age, gender, smoking status, cell-type
composition). Thus, we can compare our method result with result of method
considering explicitly these variables cite:Rahmani_2016,Zou_2014. We retrieve
the RA data from Gene Expression Omnibus (GEO) database (accession number
GSE42861). Following cite:Zou_2014 we filtered out site if its average probe
$\beta$ value was above 0.8 are below 0.2. Then, the $\beta$ values was centered
and normalized with standard deviation.

**** Association study of genetic variants with Celiac disease (GWAS)

**** Association study of genetic variants with climatic data (EAS)

* Results
* Discussion
* Figures and tables
** Numerical validation

