% use option [preprint] to remove info line at bottom
% journal options: aop,aap,aos,aoas,ssy
% natbib option: authoryear
\documentclass[12pt,aoas]{imsart} %% to review
%%\documentclass[aoas]{imsart}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% Package
\usepackage{amssymb}
\usepackage{verbatim}
%\usepackage[francais,english]{babel}
\usepackage{graphicx}
\usepackage{lineno}
\RequirePackage{natbib}
\usepackage{hyperref}
\usepackage{algorithm2e}
\RequirePackage{amsthm,amsmath}
\usepackage{color}
%\usepackage[latin1]{inputenc}
%\usepackage[table]{xcolor}

% provide arXiv number if available:
%\arxiv{arXiv:0000.0000}

% put your definitions there:
\startlocaldefs
\numberwithin{equation}{section}
\theoremstyle{plain}
\newtheorem{thm}{Theorem}[section]
\theoremstyle{definition}
\newtheorem{algo}{Algorithm}[section]
\endlocaldefs
\DeclareMathOperator*{\argmin}{arg\,min}


\begin{document}

%% To review
\baselineskip 0.8cm
\linenumbers
%%

\begin{frontmatter}

% "Title of the paper"
\title{Fast inference of individual admixture coefficients using geographic data}
\runtitle{Fast inference of individual admixture coefficients}

\begin{aug}
\author{\fnms{Kevin} \snm{Caye}\thanksref{t1}\ead[label=e1]{kevin.caye@imag.fr}},
\author{\fnms{Flora} \snm{Jay}\thanksref{t2}\ead[label=e2]{flora.jay@lri.fr}},
\author{\fnms{Olivier} \snm{Michel}\thanksref{t1}\ead[label=e3]{olivier.michel@gipsa-lab.grenoble-inp.fr}},
\and
\author{\fnms{Olivier} \snm{Fran\c cois}\thanksref{t1}
\ead[label=e4]{olivier.francois@imag.fr}}

%% \thankstext{m1}{This work has been partially supported by the LabEx PERSYVAL-Lab (ANR-11-LABX-0025-01) funded by the French program Investissement d\rq{}Avenir}
%% \thankstext{m2}{Olivier Fran\c cois acknowledges support from Grenoble INP and from the Agence Nationale de la Recherche, project AFRICROP ANR-13-BSV7-0017}
\runauthor{K. Caye et al.}

\affiliation{Universit\'e Grenoble-Alpes\thanksmark{t1} and Universit\'e Paris-Sud\thanksmark{t2}}

\address{Kevin Caye and Olivier Fran\c cois\\
Universit\'e Grenoble-Alpes\\
Centre National de la Recherche Scientifique\\ 
TIMC-IMAG UMR 5525\\
Grenoble, 38042, France\\
\printead{e1}\\
\phantom{E-mail:\ }\printead*{e4}}

\address{Flora Jay\\
Université Paris-Sud\\
Université Paris-Saclay\\
Laboratoire de Recherche en Informatique UMR 7206\\
CNRS UMR 8623\\
Orsay, 91400, France\\
\printead{e2}}

\address{Olivier Michel\\
Universit\'e Grenoble-Alpes\\
Centre National de la Recherche Scientifique\\ 
GIPSA-lab UMR 5216\\
Grenoble, 38042, France\\
\printead{e3}}


\end{aug}

\begin{abstract}
Accurately evaluating the distribution of genetic ancestry across geographic space is one of the main questions addressed by evolutionary biologists. This question has been commonly addressed through the application of Bayesian estimation programs allowing their users to estimate individual admixture proportions and allele frequencies among putative ancestral populations. Following the explosion of high-throughput sequencing technologies, several algorithms have been proposed to cope with computational burden generated by the massive data in those studies. In this context, incorporating geographic proximity in ancestry estimation algorithms is an open statistical and computational challenge. In this study, we introduce new algorithms that use geographic information to estimate ancestry proportions and ancestral genotype frequencies from population genetic data. Our algorithms combine matrix factorization methods and spatial statistics to provide estimates of ancestry matrices based on least-squares approximation.  We demonstrate the benefit of using spatial algorithms through extensive computer simulations, and we provide an example of application of our new algorithms to a set of spatially referenced samples for the plant species {\it Arabidopsis thaliana}.  Without loss of statistical accuracy, the new algorithms exhibit runtimes that are much shorter than those observed for previously developed spatial methods. Our algorithms are implemented in the {\tt R} package, {\tt tess3r}, which is available from \url{https://github.com/BioShock38/TESS3_encho_sen}. 
\end{abstract}

\begin{keyword}
\kwd{Ancestry Estimation Algorithms}
\kwd{Genotypic Data}
\kwd{Geographic Data}
\kwd{Fast Algorithms}
\end{keyword}

\end{frontmatter}


\input{intro.tex}
\input{method.tex}
\input{results.tex}
\input{discussion.tex}

\appendix

\input{appendix.tex}

\newpage

\section*{Acknowledgements}
This work has been partially supported by the LabEx PERSYVAL-Lab (ANR-11-LABX-0025-01) funded by the French program Investissement d\rq{}Avenir. Olivier Fran\c cois acknowledges support from Grenoble INP and from the Agence Nationale de la Recherche, project AFRICROP ANR-13-BSV7-0017.


% AOS,AOAS: If there are supplements please fill:
%\begin{supplement}[id=suppA]
%  \sname{Supplement A}
%  \stitle{Title}
%  \slink[doi]{10.1214/00-AOASXXXXSUPP}
%  \sdatatype{.pdf}" 
%  \sdescription{Some text}
%\end{supplement}

\input{references.tex}
\input{tables_figures.tex}

\end{document}
