* Revisions
  Recu le [2017-03-02 Jeu].
  
** Mail

   Dear Olivier François,

   Thank you for submqitting your paper "Fast inference of individual admixture
   coefficients using geographic data" for possible publication in Annals of
   Applied Statistics. It has now been carefully reviewed and my decision is:
   Revision required.

*** My comments are the following:
    The reviewers all agree that this paper is an interesting methodological
    contribution to the field of inference of population structure. There were
    four main revisions that should be made to the work according to the
    reviewer comments.
**** First, 
     slightly more space should be devoted to connecting the new work with the
     related work -- the related work is very clearly written, but the exact
     methods proposed here and the inclusion of geographic data should be
     motivated a bit more carefully and put in the perspective of the related
     work.
***** TODO a faire
        Pk on fait ca par rapport au ancienne version de TESS3. Pk faire une new
        version de tess3
**** Second, 
     the selection of hyper parameters was not well discussed, and two of the
     reviewers would rather those details be included in the paper to have it
     self-contained.
***** TODO a faire
      discuter plus la validation croisée. Pas mettre en reference.
**** Third, 
     related, R1 asks about model misspecification and how to evaluate the
     impact of incorrect priors on the results --- I agree that this point
     should be addressed adequately.
***** TODO a faire
      Test pour l'ibd, on le voit dans le variogram si il n'y a pas de d'interet
      du spatial. Il y a une grosse litérature sur ces tests ! 

**** Fourth, 
     the results should be expanded to a larger data set. There were a handful
     of minor comments that should also be addressed.


     In addition to these comments you may also find review reports posted on EJMS.

***** TODO a faire
      Prendre le AT 1001 genome. Data set est propre. 
      *Rmk* : On peut pas le faire sur des humains, on a pas les coord
      geographique, ca serait pas anonime.
*** To submit your revision,
    please log in to EJMS and submit it as a revised file to original
    submission. Please also include a detailed description of how you addressed
    all the points raised by the reviewers.

*** IMPORTANT NOTICE CONCERNING FIGURES: 

    Printing figures in color adds significantly to the production cost of the
    journal. While color may be used in the online publication, we will use
    color in the printed version only when essential to the display. Please use
    dashed/dotted lines or symbols where possible and avoid referring to colors
    in the text and the figure caption.

*** other

    If you have been asked to modify the title of your submission, or if the
    order of author names has changed, please contact Geri Mattson at
    mattsonpublishingservices@comcast.net so that the submission’s metadata can
    be updated.

    Thank you for considering The Annals of Applied Statistics as a venue for your work.

    Sincerely,
    Edoardo M. Airoldi
    Editor, The Annals of Applied Statistics
    
    
    Submission URL: https://www.e-publications.org/ims/submission/AOAS/
    
    Title:
    Fast inference of individual admixture coefficients using geographic data
    
    Authors:
    Kevin Caye, Flora Jay, Olivier Michel, Olivier François
    
    Abstract: Accurately evaluating the distribution of genetic ancestry across
    geographic space is one of the main questions addressed by evolutionary
    biologists. This question has been commonly addressed through the
    application of Bayesian estimation programs allowing their users to estimate
    individual admixture proportions and allele frequencies among putative
    ancestral populations. Following the explosion of high-throughput sequencing
    technologies, several algorithms have been proposed to cope with
    computational burden generated by the massive data in those studies. In this
    context, incorporating geographic proximity in ancestry estimation
    algorithms is an open statistical and computational challenge. In this
    study, we introduce new algorithms that use geographic information to
    estimate ancestry proportions and ancestral genotype frequencies from
    population genetic data. Our algorithms combine matrix factorization methods
    and spatial statistics to provide estimates of ancestry matrices based on
    least-squares approximation. We demonstrate the benefit of using spatial
    algorithms through extensive computer simulations, and we provide an example
    of application of our new algorithms to a set of spatially referenced
    samples for the plant species Arabidopsis thaliana. Without loss of
    statistical accuracy, the new algorithms exhibit runtimes that are much
    shorter than those observed for previously developed spatial methods. Our
    algorithms are implemented in the R package, tess3r, which is available from
    https://github.com/BioShock38/TESS3_encho_sen.

** [[file:Revisions/AOAS1610-012R1R1.txt][R1]]
*** Intro
    Inferring individual ancestry (IA) from geontype data is an important
    problem in population genetics that has received much attention from both
    statistics and genetics communities. Caye et al. focus on the IA estimation
    problem in the setting where geographic data is available. They cast this
    problem as a regularized matrix factorization problem. The goal is to find Q
    and G matrices that reside in a convex set and approximate the genotype
    matrix. The requirement that geographically proximal individuals have
    similar IA parameters enforces a regularization on the solution. The authors
    explore two algorithms to this convex optimization problem: one based on
    alternating quadratic programming (AQP) and the second based on alternating
    projected least squares (APLS). The latter is shown to provide statistically
    accurate estimates while being computationally efficient on simulated data.
    
    
    The paper proposes a novel formulation and approach to incorporate spatial
    information for estimating IA. This model could be useful in applications
    where geographic locations are available along with genetic data. I think
    the paper represents an interesting applied statistics work. However, I have
    some comments that I would like the authors to address -- specifically,
    related to their choice of regularizer, model misspecification and empirical
    comparisons.

*** Comments:
    
**** 1.  
     While it is clear that spatial information can naturally be incorporated as
     a regularizer, it is not clear what the motivation is for the specific
     choice of regularizer. For example, it is intuitively not clear why the
     regularizer is inversely proportional to K and lambda_max.

     Further, if I decide to choose the regularizer coefficient by
     cross-validation, does it matter if the regularizer is scaled by parameters
     such as K,lambda_max as long as I search over a large rage of values of the
     regularizer coefficient ?

     Given that this is the central aspect of the paper, I would like the
     authors to provide intuition for their model choice including the choice of
     regularizer.

***** TODO a faire
     Donner une intuition.  

**** 2. The empirical assessment can be improved. 

***** a) 
      One of the concerns is that the simulations appear to assume that the true
      locations are known. I would like to know how correlated the IA estimates are
      with location in the simulations. How does the performance improvement relative
      to a method that does not use spatial information change if the locations are
      noisy so that the correlation between IA estimates and location is lower.

****** TODO a faire
       Une simu : on va bruité les coordonnées géographique. On regarde comment
       le bruit sur les coordonnées influ sur l'esimation de Q. On peut faire
       varier la variance du bruit. 
***** b) 
      A second and more important concern is that it is unclear how the model
      performs in instances where genetics and geography do not correlate. For
      example, many of the instances of large-scale admxiture involve population
      migration that results in relatedness between populations that are
      separated by large genetic distances. Consider, African-Americans that are
      admixed between African and European populations. In terms of location,
      African-Americans are located in the US which is not proximal to ancestral
      Africans or Europeans. IT is unclear how the inferences would change in
      this setting.
****** TODO a faire 
       Se verifie avec les test d'autocorélation spatial, le variogram etc.
       C'est de la validation des hypothèses. 

       *MAIS* pour nous la VC ne marche pas

       *Une simu*: prendre du 1000 genomes (européen affricain et
       afro-americain) et leur donner des coords geographique et voir ce qui se
       passe (comment ca degrade par rapport à NMF). C'est un peu moins bien,
       mais faut faire des hypothèse a un moment !!
***** c) 
      An interesting question that would point to the utility of these spatial
      models is to ask how approximate or noisy does the location information
      need to be to obtain an advantage over models that do not use spatial
      information. This would be an interesting quantity that could strengthen
      the appeal of the current study.
****** TODO a faire
       Repondu, par le graphe de a)
***** d) 
      The authors should also compare to other spatially explicit methods for
      inferring IA. e.g. SpaceMix (Bradburd et al. 2015). These methods jointly
      estimate IA as well as geographic coordinates in a Bayesian framework.
****** TODO a faire
       SpaceMix est pop based ? On les citera.
** [[file:Revisions/AOAS1610-012R1R2.pdf][R2]]
*** Intro
    The authors propose an extension to their tess3 software to allow spatial
    coordinates of samples to be used to smooth local estimates of ancestry
    proportions. They use a matrix factorization approximation to the STRUCTURE
    model, which they have previously shown to give comparable results at
    reduced computational cost. Spatial smoothness in the ancestry proportions
    is attained using a Gaussian kernel whose length scale is estimated offline.
    Two optimization approaches are proposed: the first using alternating
    quadratic programming which is guaranteed to obtain a local optimum
    (strictly critical point) of the objective, the second using a heuristic
    optimize-and-project scheme which gives very comparable empirical
    performance at significantly reduced computational cost. On simulated data
    with K=2 admixed ancestral populations leveraging the spatial information is
    shown to improve estimation of the original ancestral frequencies and
    ancestry proportions. On a N=1000 dataset of A. thaliana across Europe the
    method is applied to show a distribution of multiple populations across
    Europe, and to detect candidate SNPs under selective pressure.

*** The paper is generally clearly written with an appropriate level of detail. There are some important details which are deferred to references, in particular:
   - the cross-validation scheme/objective used for choosing K
   - the variogram approach for choosing sigma
   - how SNPs are tested as being outliers under selective pressure
   It's perhaps only a personal preference but since these are key, non-
   standard steps in the analysis it would be good if they were at least
   described in the supplement so that the paper is more self-contained.

**** TODO a faire
     Expliquer ce qu'on a mis en ref.
*** Some prior work which should probably be cited:
    - Fast spatial ancestry via flexible allele frequency surfaces. Rañola JM1,
      Novembre J1, Lange K. Bioinformatics 2014.
      https://www.ncbi.nlm.nih.gov/pubmed/25012181. This method smooths both
      latent allele frequencies and allocation proportions but using a grid/pixel
      based random field approach which I assume is more computationally
      expensive than tess3r. The setup is somewhat different but a quantitative
      comparison might still be possible? Code is available in the OriGen R
      package.
    - Novel probabilistic models of spatial genetic ancestry with applications to
      stratification correction in genome-wide association studies. Anand
      Bhaskar, Adel Javanmard, Thomas A. Courtade, David Tse
      https://arxiv.org/abs/1610.07306. The problem setup between this ("GAP")
      and the current paper is quite different: GAP estimates spatial coordinates
      of individuals given their genotype data, and so should be grouped with the
      citations on lines 79-80, page 3.
**** TODO a faire
     a voir ce c'est. 
*** An analysis of at least one human dataset,  
    the Simons diversity panel being one interesting recent possibility, would
    add significantly to the paper and given the impressive run-times of the
    method presumably wouldn't be difficult to do.
**** TODO a faire
     Le pb c'est les coord spatial pour les humains ! On met AT 1001 genome,
     comme ca on fait des simunaltions pour LFMM. 

     En fait si le dataset est pas mal : [[https://www.simonsfoundation.org/life-sciences/simons-genome-diversity-project-dataset/][Simons diversity dataset]]. On va filter
     la maf et faire une belle carte (si il y a pas de pop admixed le spatial va
     renforcer le clustering)
*** I've annotated minor corrections/suggestions on the manuscript itself, hopefully attached.
** [[file:Revisions/AOAS1610-012R1R3.txt][R3]]
*** Intro
   In this paper, Caye et al. present the newest iteration of their tess
   algorithm, which constructs an STRUCTURE-like mixed membership model while
   taking the spatial origin of data into account. This is a highly relevant
   problem, as spatial awareness has the potential to increase power, and gives
   more sensible answers when sampling is highly uneven.

   The main purpose of this paper is the presentation of two new algorithms, AQP
   and APLS, that both ofter fast runtimes. The reason why a standard EM cannot
   be used for the present problem is that the spatial awareness enters the
   model in the from a penalty matrix, without explicitly constructing a model.

   As someone unfamiliar with the algorithms presented here, the details
   presented in the paper are enough to follow the basic ideas behind the two
   minimization procedures,
     

*** APLS
   The APLS aogorithm proceeds by first updating each locus individually
   (assuming knowledge of each individual (the Q matrix) unconstrained, and then
   the constraints are enforced by a projection onto the relevant subspaces. As
   someone interested in this approach without too much knowledge in the field,
   I found the description to be lacking, as I was neither informed on how the
   implementation works, nor how the approximation is justified. Spending some
   more space on on what is the major innovation of the project could greatly
   enhance this paper.
**** TODO a faire
     Description plus verbale d'APLS: idée clées.
   
*** Simulations
    The simulation study accompanying the paper is adequate, and convincing that
    the implementation is correct and appropriate. They empirically show that the
    approximations arrive at a solution without any substantial change in error,
    and show that, under the assumed model, that adding space as a covariate
    increases power and reduces error. The underlying problem that is not
    addressed, is what "homogeneity" assumptions are made regarding the spatial
    patterns. I would expect that for populations whose genetic make-up is only
    loosely associated with space, that there is some point where a non-spatial
    algorithm might perform better. This may also be the reason why tess is used
    a lot less than structure/admixture in empirical studies, since the apparent
    assumption of strong spatial structure is not always that easy to make. One
    set of simulations to address that may be to repeat the analysis of fig 1
    where individuals are assigned locations at random. However, since the paper
    is highly technical and empiricists are not likely to be the target audience,
    this may not be the appropriate place for this.
**** TODO a faire    
     Encore un fois c'est le variogramm les test d'autocorélation spatials. Et
     ca serais résolu par le CV. On va mettre les graphes qu'on a fait pour R1.

*** AT
    The application to Arabidopis lacks a comparison point, it would have been
    interesting to compare the result with sNMF or earlier versions of TESS. One
    interesting point, for example, is that the ancestry coefficients in Fig 6B
    appear to be less peaked than in e.g. the data from the Francois et al. 2008
    paper, is this a function of the larger data set or the new algorithm?
    Finally, figure 6A has some extrapolation artefacts that should be corrected.
    Regions in Anatolia and Scandinavia appear to not-have any samples, but are
    assigned clusters from different regions. I assume this is a weird
    tail-behaviour in the spatial smoothing algorithm.
**** TODO a faire
     - enlever l'artefact en turquie.

  Overall, I think this is a solid paper, but the presentation of the main
  algorithms could be a bit more detailed, if not in the main text, in a
  supplementary technical reference.
** On résume à faire
*** TODO experiments
    - [ ] on reprend les simulations et on bruite les coords. graph RMSE(Q) x
      sigma(bruit) x regularization param 
    - [ ] CV si ca marche ! 
    - [ ] simu 1000 genome (European Africain et Afro americain). Que donne snmf et
      tess3r. Le fichier est énorme, on va faire un LF prunning et un filtrage
      par la maf ! 
    - [ ] simons avec la maf 5%
*** TODO implémentation
    - [ ] un fonction prédict sur un indiv pas vu ! Qui pourrait servir pour la
      cross validation.
* 2017
** 2017-03 mars
*** 2017-03-07 mardi
**** TODO Réorganisation
     [2017-03-07 mar. 14:14]

     - [ ] fork tess3r
     - [ ] script -> package
     - [ ] Makefile 

     Pour le moment j'ai fait un gros copy paste de map.R ...
     To be continued.

***** Factoriser du code entre les packages pour la thèse ? 
      - benchmark function
      - export function
      - shinny function
      - notebook 
      a reflechir
